packages/matplotlib/patches/hardcoded-font-cache.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/lib/matplotlib/font_manager.py b/lib/matplotlib/font_manager.py
+--- a/lib/matplotlib/font_manager.py
++++ b/lib/matplotlib/font_manager.py
+@@ -1394,8 +1394,7 @@ def is_opentype_cff_font(filename):
+         return False
+ 
+ 
+-_fmcache = os.path.join(
+-    mpl.get_cachedir(), 'fontlist-v{}.json'.format(FontManager.__version__))
++_fmcache = os.path.join(os.path.dirname(__file__), "fontlist.json")
```

Comment:
Do you understand why we need this? Couldn't we just use the version of FontManager that's used in matplotlib?

Reply:
You mean the `fontlist.json` file ? Removing it will lead to a full rebuild which may take some time. See below.

```python

def _rebuild():
    global fontManager
    _log.info("Generating new fontManager, this may take some time...")
    fontManager = FontManager()
    json_dump(fontManager, _fmcache)


try:
    fontManager = json_load(_fmcache)
except Exception:
    _rebuild()
else:
    if getattr(fontManager, '_version', object()) != FontManager.__version__:
        _rebuild()
    else:
        _log.debug("Using fontManager instance from %s", _fmcache)
```

I'll investigate to see if it really take some time.

Reply:
Between about 1.5 second to build the font cache and save it to virtual FS. What do we do?

Reply:
Doing this a see big differences between generated version and the one of this PR. If we choose to keep it on the Pyodide I should update it.

Reply:
Thanks for investigating, let's keep the vendored version in Pyodide and update it.